cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/local/LocalStorageClient.java

```diff
@@ -57,19 +61,19 @@ public synchronized void init() throws URISyntaxException, IOException {
               .getEndpoint()
               .startsWith(DEFAULT_ENDPOINT),
           "Storage properties endpoint was misconfigured for: " + LOCAL_TYPE.getValue());
-      try {
-        uri =
-            new URI(
-                storageProperties.getTypes().get(LOCAL_TYPE.getValue()).getEndpoint()
-                    + storageProperties.getTypes().get(LOCAL_TYPE.getValue()).getRootPath());
-      } catch (URISyntaxException e) {
-        throw new IllegalArgumentException(
-            "Storage properties 'endpoint', 'rootpath' was incorrectly configured for: "
-                + LOCAL_TYPE.getValue(),
-            e);
-      }
+      endpoint = storageProperties.getTypes().get(LOCAL_TYPE.getValue()).getEndpoint();
+      rootPath = storageProperties.getTypes().get(LOCAL_TYPE.getValue()).getRootPath();
     } else {
-      uri = new URI(DEFAULT_ENDPOINT + DEFAULT_ROOTPATH);
+      endpoint = DEFAULT_ENDPOINT;
+      rootPath = DEFAULT_ROOTPATH;
+    }
+    try {
+      uri = new URI(endpoint + rootPath);
```

Comment:
Nit: Given we plan to use FileSystem for Local filesystem, should we construct https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/Path.html instead?

Reply:
`uri` ends up as an argument to `FileSystem.get`, I couldn't find a good way to create a file system with a path.
![image](https://github.com/linkedin/openhouse/assets/6441597/e92f5f93-ab0b-4e0c-b784-c781109cf8d8)